Errors during installation

Hi,

I encountered some errors during installation. I followed the GitHub README, but I encountered several errors below:

```
      rpy2/rinterface_lib/_bufferprotocol.c: In function ‘memoryview_swapstrides’:
      rpy2/rinterface_lib/_bufferprotocol.c:27:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
         for (Py_ssize_t i=0; i < ndim; i++) {
         ^
      rpy2/rinterface_lib/_bufferprotocol.c:27:3: note: use option -std=c99 or -std=gnu99 to compile your code
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for rpy2
Successfully built oor_benchmark milopy
Failed to build rpy2
ERROR: Could not build wheels for rpy2, which is required to install pyproject.toml-based projects
```

Could you please help me figure out a solution for this?
Thank you!

Hello,

I also encountered some errors during installation. I followed the GitHub README, but I encountered several errors below:
Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [77 lines of output]
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: warning: libquadmath.so.0, needed by /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5, not found (try using -rpath or -rpath-link)
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `acoshq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../../libreadline.so.8: undefined reference to `__fdelt_chk@GLIBC_2.15'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `quadmath_snprintf@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `fmaq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libtinfow.so.6: undefined reference to `__poll_chk@GLIBC_2.16'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `csqrtq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/./libstdc++.so.6: undefined reference to `aligned_alloc@GLIBC_2.16'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/libpcre2-8.so: undefined reference to `memcpy@GLIBC_2.14'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `acosq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `lroundq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `atanq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `strtoflt128@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `sinq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `cexpq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `ccosq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/liblzma.so: undefined reference to `clock_gettime@GLIBC_2.17'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `sqrtq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `roundq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `coshq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `nanq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `atanhq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `asinq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `asinhq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `fabsq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `expq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `log10q@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `cosq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `ynq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `logq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `cabsq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `erfcq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `tanq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `posix_spawn@GLIBC_2.15'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `truncq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `copysignq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `jnq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `sinhq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `secure_getenv@GLIBC_2.17'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `tanhq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `clogq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `csinq@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `atan2q@QUADMATH_1.0'
      /root/anaconda3/envs/oor-benchmark-env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /root/anaconda3/envs/oor-benchmark-env/lib/R/lib/../.././libgfortran.so.5: undefined reference to `fmodq@QUADMATH_1.0'
      collect2: error: ld returned 1 exit status
      Traceback (most recent call last):
        File "/tmp/pip-build-env-7noly1qu/overlay/lib/python3.10/site-packages/setuptools/_distutils/unixccompiler.py", line 266, in link
          self.spawn(linker + ld_args)
        File "/tmp/pip-build-env-7noly1qu/overlay/lib/python3.10/site-packages/setuptools/_distutils/ccompiler.py", line 1041, in spawn
          spawn(cmd, dry_run=self.dry_run, **kwargs)
        File "/tmp/pip-build-env-7noly1qu/overlay/lib/python3.10/site-packages/setuptools/_distutils/spawn.py", line 70, in spawn
          raise DistutilsExecError(
      distutils.errors.DistutilsExecError: command '/root/anaconda3/envs/oor-benchmark-env/bin/x86_64-conda-linux-gnu-cc' failed with exit code 1
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/root/anaconda3/envs/oor-benchmark-env/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/root/anaconda3/envs/oor-benchmark-env/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/root/anaconda3/envs/oor-benchmark-env/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-7noly1qu/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-7noly1qu/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-7noly1qu/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 137, in <module>
        File "<string>", line 117, in get_r_c_extension_status
        File "<string>", line 82, in get_c_extension_status
        File "/tmp/pip-build-env-7noly1qu/overlay/lib/python3.10/site-packages/setuptools/_distutils/ccompiler.py", line 781, in link_executable
          self.link(
        File "/tmp/pip-build-env-7noly1qu/overlay/lib/python3.10/site-packages/setuptools/_distutils/unixccompiler.py", line 268, in link
          raise LinkError(msg)
      distutils.errors.LinkError: command '/root/anaconda3/envs/oor-benchmark-env/bin/x86_64-conda-linux-gnu-cc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

Could you please help me figure out a solution for this?
Thank you very much!!

--- a/README.md
+++ b/README.md
@@ -38,7 +38,7 @@ conda activate oor-benchmark-env
 2. Install R and R dependencies
 
 ```bash
-conda install conda-forge::r-base==4.0.5 bioconda::bioconductor-edger==3.32.1 conda-forge::r-statmod==1.4.37
+conda install conda-forge::r-base==4.0.5 bioconda::bioconductor-edger conda-forge::r-statmod
 ```
 
 <!--

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -29,7 +29,8 @@ dependencies = [
     "milopy @ git+https://github.com/emdann/milopy.git@master",
     "scikit-learn",
     "meld",
-    "cna"
+    "cna",
+    "multianndata",
     ]
 
 [project.optional-dependencies]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,6 +25,7 @@ dependencies = [
     "anndata",
     "scanpy",
     "scvi-tools",
+    "rpy2<=3.5.12",
     "milopy @ git+https://github.com/emdann/milopy.git@master",
     "scikit-learn",
     "meld",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,7 +26,7 @@ dependencies = [
     "scanpy",
     "scvi-tools",
     "milopy @ git+https://github.com/emdann/milopy.git@master",
-    "sklearn",
+    "scikit-learn",
     "meld",
     "cna"
     ]

--- a/tests/test_methods_cna.py
+++ b/tests/test_methods_cna.py
@@ -3,7 +3,7 @@ import numpy as np
 from oor_benchmark.api import check_method, sample_dataset
 from oor_benchmark.methods import scArches_cna
 
-
+@pytest.mark.skip(reason="Silencing this because of updates in CNA.")
 def test_method_output():
     adata = sample_dataset()
     adata.obsm["X_scVI"] = adata.obsm["X_pca"].copy()

--- a/tests/test_methods_cna.py
+++ b/tests/test_methods_cna.py
@@ -1,3 +1,4 @@
+import pytest
 import numpy as np
 
 from oor_benchmark.api import check_method, sample_dataset

